docs/guide/interactions/context_menus.rst

```diff
@@ -0,0 +1,52 @@
+:orphan:
+
+.. currentmodule:: discord
+
+.. _guide_interactions_context-menus:
+
+Context Menus Guide
+====================
+
+Context menus allow for commands to be triggered through a context menu upon right clicking a related object in the Discord UI,
+then selecting the name of the command to run from the ``Apps`` menu.
+
+.. note::
+
+    Context menus are not designed to be put inside of cogs. If you would like to use a context menu within a cog,
+    you will need to handle creating the :class:`.app_commands.ContextMenu` yourself.
+
+
+Basic Usage
+~~~~~~~~~~~~
+
+.. code-block:: python3
+
+    @app_commands.context_menu(name="User Context")
+    async def example(interaction: discord.Interaction, user: discord.User | discord.Member):
+        await interaction.response.send_message(f"Interacting with user {user}.", ephemeral=True)
+
+    @app_commands.context_menu(name="Message Context")
+    async def example(interaction: discord.Interaction, message: discord.Message):
+        await interaction.response.send_message(f"Interacting with message {message.content}.", ephemeral=True)
+
+- Context menus can be triggered from two possible sources: by right clicking on a user or server member, or by right clicking on a message.
+- The ``name`` argument of the :func:`@context_menu <.app_commands.context_menu>` decorator defines the command name that appears in the context menu.
+- The second argument can be of type :class:`.User`, :class:`.Member`, or :class:`.User` | :class:`.Member` for user context menus.
```

Comment:
It's not obvious to me reading this what the distinction between User, Member or User | Member is

Reply:
I'm not sure how to specify this one honestly. 

To my knowledge, when used in a Guild context, the arg will be a Member, and when used in a DM context, it will be a User. However, you can also get a User object for both Guild and DMs, or use the union to get either one respective to what environment you invoke the command/menu in.

Reply:
Ah right, makes a bit more sense now, not sure if there's a better way to put this, I'll leave that to someone else